pkg/git/copy.go

```diff
@@ -1,17 +1,45 @@
 package git
 
-import "path"
+import (
+	"path"
+	"strings"
+)
 
-// CopyService takes a path within a repository, and a source and destination,
-// and copies the files from within the source, to the destination.
+// CopyService takes the name of a service to copy from a Source to a Destination.
+// Source, Destination implement Walk() and are typically Repository objects.
+//
+// Only files under /services/[serviceName]/base/config/* are copied to the destination
 //
 // Returns the list of files that were copied, and possibly an error.
-func CopyService(filePath string, source Source, dest Destination) ([]string, error) {
+func CopyService(serviceName string, source Source, dest Destination) ([]string, error) {
+
+	// filePath defines the root folder for serviceName's config in the repository
+	filePath := pathForServiceConfig(serviceName)
+
 	copied := []string{}
 	err := source.Walk(filePath, func(prefix, name string) error {
-		err := dest.CopyFile(path.Join(prefix, name), name)
-		copied = append(copied, name)
-		return err
+		sourcePath := path.Join(prefix, name)
+		destPath := pathForServiceConfig(name)
+		if pathValidForPromotion(serviceName, destPath) {
+			err := dest.CopyFile(sourcePath, destPath)
+			copied = append(copied, destPath)
+			return err
+		}
+		return nil
 	})
 	return copied, err
 }
+
+// pathValidForPromotion()
+//  For a given serviceName, only files in services/serviceName/base/config/* are valid for promotion
+//
+func pathValidForPromotion(serviceName, filePath string) bool {
+	filterPath := path.Join("services", serviceName, "base/config")
```

Comment:
How about calling `pathForServiceConfig` here?

Also, "base/config" would be better split into two items, and let path.Join do the work.

Also, instead of using `path.Join` I realised this should use `filepath.Join` which would handle cross-platform better.

Reply:
Good catches - done.